pentaho-js/src/main/javascript/cdf/components/XactionComponent.js

```diff
@@ -49,7 +49,20 @@ define([
                                   " height=\"100%\"" +
                                   " width=\"100%\" />";
           var iframe = $(xactionIFrameHTML);
-          var url = XactionComponentExt.getCdfXaction(this.path, this.action, this.solution) + "&wrapper=false";
+          var actionIncluded = function(path, action) {
+            //check if path ends with action prefixed with '\' or '/'
+            return (typeof path == "string") && (typeof action == "string")
+                && (path.length > action.length)
+                && (path.lastIndexOf(action) == (path.length - action.length))
+                && ("\\/".indexOf(path.substr(-action.length-1, 1))>=0);
+          };
+          var url;
+          if (actionIncluded(myself.path, myself.action)) {
+            url = XactionComponentExt.getCdfXaction(myself.path, "", myself.solution) + "&wrapper=false";
+          } else {
+            url = XactionComponentExt.getCdfXaction(myself.path, myself.action, myself.solution) + "&wrapper=false";
+          }
+
```

Comment:
Here, I'd prefer to not repeat the longer expression, which calculates the final URL.
It could be like:
```js
var actionEf = pathIncludesAction(this.path, this.action) ? "" : this.action;

var url = XactionComponentExt.getCdfXaction(this.path, actionEf, this.solution) +
     "&wrapper=false";
```

Additionally, note that it is not necessary to use the `myself` variable, instead of `this`, as this code lies _directly_ in the body of the `update` method.